### DISCUSSION TITLE
Changing / Adding a Header Value?

### DISCUSSION THREAD

**Discussion by mwyres** [zbateson/mail-mime-parser#242] [General]
Maybe I'm not reading in the right spot, or maybe I've not consumed enough caffeine this morning... :)

...but does this library provide a way to set/change the value of an existing header? and add new headers - (eg: X-Something)?

**Comment by mwyres** [comment]
...just to add where I'm at - (yes, I know "setHeaderValue" doesn't appear to be a thing) - I'm just looking for a function that achieve what it suggests:

@require_once("./vendor/autoload.php");
use ZBateson\MailMimeParser\MailMimeParser;
use ZBateson\MailMimeParser\Message;
use ZBateson\MailMimeParser\Header\HeaderConsts;

$string = @file_get_contents("./spool/example1.eml");

$mailParser = new MailMimeParser();

$message = $mailParser->parse($string,FALSE);

var_dump($message->getSubject());
var_dump($message->getHeader('From')->getPersonName());
var_dump($message->getHeader('From')->getEmail());
var_dump($message->getHeader('X-Michael-Test')->getRawValue());

$test = $message->setHeaderValue("X-Michael-Test","test value");

**Reply by zbateson** [reply]
Hi @mwyres  -- what you're looking for is [setRawHeader](https://mail-mime-parser.org/api/3.0/classes/ZBateson-MailMimeParser-Message-IMimePart.html#method_setRawHeader)

The reason it's 'raw' is because no formatting, etc... is done for you, you must provide it formatted.

**Reply by mwyres** [reply]
Brilliant...the "raw" piece is what I missed reading the doco.  Working perfectly - thank you!
